src/Modules/OrchardCore.Commerce/Controllers/PaymentController.cs

```diff
@@ -36,9 +40,18 @@ public async Task<IActionResult> Index()
         return View();
     }
 
-    [Route("success")]
-    public IActionResult Success() =>
-        View();
+    [Route("success/{orderId}")]
+    public async Task<IActionResult> Success(string orderId)
+    {
+        var order = await _contentManager.GetAsync(orderId);
+
+        if (order != null)
```

Comment:
If the order is `null` it should `return NotFound();`. Otherwise the view will error out.
![image](https://user-images.githubusercontent.com/4007293/186643943-807cd10c-8ea2-48d8-ac01-97878114e6e1.png)